### PR TITLE
[size-report] support github actions

### PR DESCRIPTION
**Pull request by bukepo**

### CODE REVIEW

File: script/check-size [hunk 1]
```diff
@@ -27,47 +27,77 @@
...
+
+    # Fetch base commit in case it is not present
+    git fetch --depth 1 --no-recurse-submodules origin "${MERGE_BASE_SHA}"
+
+    export OT_MERGE_BASE_SHA
 
     # pull request
-    OPENTHREAD_FLAGS="BORDER_AGENT=1 BORDER_ROUTER=1 CHANNEL_MANAGER=1 CHANNEL_MONITOR=1 CHILD_SUPERVISION=1 COAP=1 COAPS=1 COMMISSIONER=1 DHCP6_CLIENT=1 DHCP6_SERVER=1 DIAGNOSTIC=1 DISABLE_DOC=1 DNS_CLIENT=1 ECDSA=1 FULL_LOGS=1 JAM_DETECTION=1 JOINER=1 LINK_RAW=1 MAC_FILTER=1 MTD_NETDIAG=1 SERVICE=1 SLAAC=1 SNTP_CLIENT=1 TIME_SYNC=1 UDP_FORWARD=1"
+    local flags=(
+    "BORDER_AGENT=1"
```
Comment (jwhui) [inline script/check-size:60]: nit: add 1-level indent?

**Comment by abtink** [comment]
This is great. Thanks @bukepo.
With this I think we will run the size-report as GH action on all branches.  Do you happen to know if there is way to see the size report result on user dev branch before submitting it as a PR (e.g., from build/run [logs](https://github.com/openthread/openthread/pull/4825/checks?check_run_id=585094318) on size-report action  maybe?)

**Comment by bukepo** [comment]
@abtink I think it's doable and we can do it in a separate PR?

**Comment by abtink** [comment]
@bukepo SGTM. Thanks. I think would be very helpful.

**Pull request merged**